pkg/webhook/elasticquota/plugin_check_quota_meta_validate.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package elasticquota
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/scheduling/v1alpha1"
+)
+
+type QuotaMetaChecker struct {
+	client.Client
+	*admission.Decoder
+	QuotaTopo *quotaTopology
+}
+
+var (
+	quotaMetaCheck = &QuotaMetaChecker{
+		QuotaTopo: nil,
+	}
+)
+
+func NewPlugin(decoder *admission.Decoder, client client.Client) *QuotaMetaChecker {
+	quotaMetaCheck.Client = client
+	quotaMetaCheck.Decoder = decoder
+	if quotaMetaCheck.QuotaTopo == nil {
+		quotaMetaCheck.QuotaTopo = NewQuotaTopology(client)
+	}
+	return quotaMetaCheck
+}
+
+func (c *QuotaMetaChecker) Admit(ctx context.Context, req admission.Request, obj runtime.Object) error {
+	klog.V(5).Infof("start to admit quota: %+v", obj)
+	if req.Operation != v1.Create {
+		return nil
+	}
+
+	quotaObj := obj.(*v1alpha1.ElasticQuota)
+	return c.QuotaTopo.AddQuota(quotaObj)
+}
+
+func (c *QuotaMetaChecker) Validate(ctx context.Context, req admission.Request, obj runtime.Object) error {
+	quotaObj := obj.(*v1alpha1.ElasticQuota)
+
+	klog.V(5).Infof("start to validate quota :%+v", quotaObj)
+
+	switch req.AdmissionRequest.Operation {
```

Comment:
As discuessed, here should validate OnAdd

Reply:
fixed